File: Java/NameGen.java
import java.util.Random;
import java.io.BufferedReader;
import java.io.File;
import java.io.FileReader;
import java.io.IOException;
import java.util.ArrayList;

public class NameGen{

  private String path1, path2, path3, path4;

  private static ArrayList<String> firstBank;
  private static ArrayList<String> lastBank;
  private static ArrayList<String> placeBank;
  private static ArrayList<String> teamBank;

  Random rand;

  public NameGen(){
    rand = new Random();
    path1 = "./nameGen/res/first.txt";
    path2 = "./nameGen/res/last.txt";
    path3 = "./nameGen/res/team.txt";
    path4 = "./nameGen/res/place.txt";
    populate();
  }

  public NameGen(String inDirectory){
    rand = new Random();
    path1 = inDirectory + "first.txt";
    path2 = inDirectory + "last.txt";
    path3 = inDirectory + "team.txt";
    path4 = inDirectory + "place.txt";
    populate();
  }

  private void populate(){
    if(firstBank != null){
      //Already loaded
      return;
    }

    firstBank = new ArrayList<String>();
    lastBank = new ArrayList<String>();
    placeBank = new ArrayList<String>();
    teamBank = new ArrayList<String>();

    try{
      BufferedReader inFile1 = new BufferedReader(new FileReader(path1));
      BufferedReader inFile2 = new BufferedReader(new FileReader(path2));
      BufferedReader inFile3 = new BufferedReader(new FileReader(path3));
      BufferedReader inFile4 = new BufferedReader(new FileReader(path4));
      String temp = "";

      if (inFile1.ready()) {
        while ((temp = inFile1.readLine()) != null) {
          firstBank.add(temp);
        }
          inFile1.close();
      } else {
        System.out.println("Unable to open input file");
      }

      if (inFile2.ready()) {
        while ((temp = inFile2.readLine()) != null) {
          lastBank.add(temp);
        }
          inFile2.close();
      } else {
        System.out.println("Unable to open input file");
      }

      if (inFile3.ready()) {
        while ((temp = inFile3.readLine()) != null) {
          teamBank.add(temp);
        }
          inFile3.close();
      } else {
        System.out.println("Unable to open input file");
      }

      if (inFile4.ready()) {
        while ((temp = inFile4.readLine()) != null) {
          placeBank.add(temp);
        }
          inFile4.close();
      } else {
        System.out.println("Unable to open input file");
      }
    }catch(IOException e){
      e.printStackTrace();
    }

    firstBank.trimToSize();
    lastBank.trimToSize();
    teamBank.trimToSize();
    placeBank.trimToSize();
  }

  public String genFirst(){
    int temp = rand.nextInt(firstBank.size());
    return firstBank.get(temp);
  }

  public String genLast(){
    int temp = rand.nextInt(lastBank.size());
    return lastBank.get(temp);
  }

  public String genFull(){
    return genFirst() + " " + genLast();
  }

  public String genPlace(){
    int temp = rand.nextInt(placeBank.size());
    return placeBank.get(temp);
  }

  public String genTeam(boolean withPlace){
    int temp = rand.nextInt(teamBank.size());
    String hold = teamBank.get(temp);
    if (withPlace) {
      return genPlace() + " " + hold;
    } else {
      return hold;
    }
  }

}

File: C++/NameGen.cpp
/**
  NameGen.cpp
  Name generating object meant to be portable between projects

  @author Willis Hand
  @version 2.0
  @copyright 2017 Willis Hand
*/
#include "NameGen.hpp"

vector<string> NameGen::firstBank;
vector<string> NameGen::lastBank;
vector<string> NameGen::placeBank;
vector<string> NameGen::teamBank;

NameGen::NameGen() {
  path1 = "./nameGen/res/first.txt";
  path2 = "./nameGen/res/last.txt";
  path3 = "./nameGen/res/team.txt";
  path4 = "./nameGen/res/place.txt";
  populate();
}

NameGen::NameGen(string inDirectory){
  path1 = inDirectory + "first.txt";
  path2 = inDirectory + "last.txt";
  path3 = inDirectory + "team.txt";
  path4 = inDirectory + "place.txt";
  populate();
}

void NameGen::populate(){
  if(firstBank.size() > 0){
    //printf("Already loaded\n");
    return;
  }
  ifstream inFile1(path1);
  ifstream inFile2(path2);
  ifstream inFile3(path3);
  ifstream inFile4(path4);
  string temp;

  if (inFile1.is_open()) {
      while (getline(inFile1, temp)) {
      firstBank.push_back(temp);
    }
      inFile1.close();
    } else {
      printf("Unable to open input file\n");
    }

  if (inFile2.is_open()) {
      while (getline(inFile2, temp)) {
      lastBank.push_back(temp);
    }
      inFile2.close();
    } else {
      printf("Unable to open input file\n");
    }

  if (inFile3.is_open()) {
      while (getline(inFile3, temp)) {
      teamBank.push_back(temp);
    }
      inFile3.close();
    } else {
      printf("Unable to open input file\n");
    }

  if (inFile4.is_open()) {
      while (getline(inFile4, temp)) {
      placeBank.push_back(temp);
    }
      inFile4.close();
    } else {
      printf("Unable to open input file\n");
    }

  firstBank.shrink_to_fit();
  lastBank.shrink_to_fit();
  teamBank.shrink_to_fit();
  placeBank.shrink_to_fit();
}

string NameGen::genFirst() {
  int temp = rand() % firstBank.size();
  return firstBank[temp];
}

string NameGen::genLast() {
  int temp = rand() % firstBank.size();
  return lastBank[temp];
}

string NameGen::genFull() {
  return genFirst() + " " + genLast();
}

string NameGen::genPlace() {
  int temp = rand() % placeBank.size();
  return placeBank[temp];
}

string NameGen::genTeam(bool withPlace) {
  int temp = rand() % teamBank.size();
  string hold = teamBank[temp];
  if (withPlace) {
    return genPlace() + " " + hold;
  } else {
    return hold;
  }
}
